lib/lib/privacyMode.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from "react"
+import {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useState,
+} from "react"
+
+export const PrivacyModeContext = createContext<{
+  enabled: boolean
+  enable: () => void
+  disable: () => void
+  toggle: () => void
+}>({
+  enabled: false,
+  enable: () => null,
+  disable: () => null,
+  toggle: () => null,
+})
+
+export const PrivacyModeProvider: React.FC<{
+  initiallyEnabled?: boolean
+  children: ReactNode
+}> = ({ initiallyEnabled = false, children }) => {
+  const [enabled, setEnabled] = useState(initiallyEnabled)
+
+  const enable = useCallback(() => {
+    setEnabled(true)
+  }, [])
+
+  const disable = useCallback(() => setEnabled(false), [])
+
+  const toggle = useCallback(() => setEnabled((e) => !e), [])
+
+  return (
+    <PrivacyModeContext.Provider value={{ enable, disable, toggle, enabled }}>
+      {children}
+    </PrivacyModeContext.Provider>
+  )
+}
+
+export const usePrivacyMode = () => {
+  return useContext(PrivacyModeContext)
```

Comment:
The problem with this is that the return value of this can be `undefined`. A trick people generally use with this is to check for existance and raise otherwise - this way you get proper typing:

```suggestion
  let context = useContext(PrivacyModeContext)
  if(context == undefined) throw "usePrivacyMode requires wrapping the component in a PrivacyModeProvider`
  return context
```

Reply:
Oh I see, makes sense